homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -0,0 +1,394 @@
+"""
+Sensor to collect the reference daily prices of electricity ('PVPC') in Spain.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/integrations/pvpc_hourly_pricing/
+"""
+import asyncio
+from datetime import date, datetime, timedelta
+import logging
+from random import randint
+from typing import Dict, List, Optional
+
+import aiohttp
+import async_timeout
+from pytz import UTC, timezone
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import ENTITY_ID_FORMAT, PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME, CONF_TIMEOUT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.event import (
+    async_track_point_in_time,
+    async_track_time_change,
+)
+from homeassistant.helpers.restore_state import RestoreEntity
+import homeassistant.util.dt as dt_util
+
+from . import SENSOR_SCHEMA
+from .const import ATTR_TARIFF, DEFAULT_TIMEOUT, DOMAIN, TARIFFS
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(SENSOR_SCHEMA.schema)
+
+_ATTRIBUTION = "Data retrieved from api.esios.ree.es by REE"
+_PRECISION = 5
+_RESOURCE = (
```

Comment:
Please break this out into a pypi package.  Home Assistant shouldn't know about the implementation details on how the data is fetched.

Reply:
Done